### ISSUE TITLE
The conv_app received a wrong dim input when causal is set to false

### ISSUE THREAD

**Issue by ZhaoQinlao** [Yuhan-Shen/ProTAS#5]
I tried to resupplement your work, but I met an error when using causal=False as my input. This may be caused by file model line 90~91
```
 self.gru_app = nn.GRU(num_f_maps, num_f_maps, num_layers=1, batch_first=True, bidirectional=not causal)
 self.conv_app = nn.Conv1d(num_f_maps, num_classes, 1)
```
The forward code in line 99~100
```
progress_out, _ = self.gru_app(out.permute(0, 2, 1))
progress_out = progress_out.permute(0, 2, 1)
```
I changed the self.conv_app to 
`self.conv_app = nn.Conv1d(num_f_maps, num_classes, 1) if causal else nn.Conv1d(num_f_maps*2, num_classes, 1)`
This worked, but I'm not sure whether it's correct.

**Comment by Yuhan-Shen** [comment]
Hi, you are correct. You can also set the output dim of gru_app as num_f_maps//2 to be consistent with the input dim of conv_app.
